daemon/cmd/daemon_main.go

```diff
@@ -1373,6 +1376,11 @@ func runDaemon() {
 	metricsErrs := initMetrics()
 
 	d.startAgentHealthHTTPService(fmt.Sprintf("localhost:%d", option.Config.AgentHealthPort))
+	if option.Config.KubeProxyReplacementHealthzBindAddr != "" {
+		if option.Config.KubeProxyReplacement == option.KubeProxyReplacementDisabled {
+			d.startAgentHealthHTTPService(fmt.Sprintf("%s", option.Config.KubeProxyReplacementHealthzBindAddr))
```

Comment:
What happens if `kube-proxy` is actually running? Will it fail?

Reply:
Yes it would fail. When kube-proxy is actually running there is an error such as 'bind: addr already in use', since kube-proxy is already creating the service with the same port.
Similar to 'HealthCheckNodePort'

Reply:
If we enable it by default, then it will break all deployments which are running with kube-proxy. Therefore, I suggest to either disable it by default or change the default port.

Reply:
Probably disabling it by default would be good idea and if users want to enable it for kube-proxy free environment they can enable it.